### ISSUE TITLE
North Lincs Council, UK Error 502 - Not Working 

### ISSUE THREAD

**Issue by Media-1** [mampfes/hacs_waste_collection_schedule#1040]
Issue:

It seems like the URL that is generated is either no longer working or incorrect. If I'm wrong, please feel free to correct me

Config: 

```
waste_collection_schedule:
    sources:
    - name: northlincs_gov_uk
      args:
        uprn: "REDCATED"
      customize:
        - type: Plastic and cardboard wheeled bin
          alias: Recycling
sensor:
    - platform: waste_collection_schedule
      name: Recycling
      details_format: upcoming
      value_template: 'in {{value.daysTo}} days'
```

Error:
```
This error originated from a custom integration.

Logger: waste_collection_schedule.source_shell
Source: custom_components/waste_collection_schedule/waste_collection_schedule/source_shell.py:136
Integration: waste_collection_schedule (documentation)
First occurred: 19:29:12 (1 occurrences)
Last logged: 19:29:12

fetch failed for source North Lincolnshire Council: Traceback (most recent call last): File "/config/custom_components/waste_collection_schedule/waste_collection_schedule/source_shell.py", line 134, in fetch entries = self._source.fetch() ^^^^^^^^^^^^^^^^^^^^ File "/config/custom_components/waste_collection_schedule/waste_collection_schedule/source/northlincs_gov_uk.py", line 33, in fetch r.raise_for_status() File "/usr/local/lib/python3.11/site-packages/requests/models.py", line 1021, in raise_for_status raise HTTPError(http_error_msg, response=self) requests.exceptions.HTTPError: 502 Server Error: Bad Gateway for url: https://m.northlincs.gov.uk/collection_dates/(REDACTED)/0/6?_=1546855781728&format=json
```

**Comment by dt215git** [comment]
Endpoint seems to have changed to:
`m.northlincs.gov.uk/bin_collections?no_collections=20&uprn=**uprn_goes_here**`

The returned data seems broadly similar to what the original script was extracting info from, but is throwing the following error so a bit more tweaking will be needed to get it working.
```bash
Unexpected UTF-8 BOM (decode using utf-8-sig): line 1 column 1 (char 0)
```

**Comment by 5ila5** [comment]
fixed by #1041

Will be available in release 1. 41.0 or now as master (git) version
